src/rules/at-root-no-redundant/index.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+const { utils } = require("stylelint");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("at-root-no-redundant");
+
+function isWithinKeyframes(node) {
+  let parent = node.parent;
+  while (parent) {
+    if (parent.type === "atrule" && parent.name === "keyframes") {
+      return true;
+    }
+    parent = parent.parent;
+  }
+  return false;
+}
+
+const messages = utils.ruleMessages(ruleName, {
+  rejected: "Unnecessary @at-root rule"
```

Comment:
If I'm not mistaken this should be `Unexpected @at-root rule`.

https://stylelint.io/developer-guide/rules#add-problem-messages

Reply:
Right, thank you!